rules/code-quality/tests/Rector/If_/MoveOutMethodCallInsideIfConditionRector/Fixture/if_condition_method_call_with_arg_on_no_assignment.php.inc

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Rector\Php\Tests\Rector\If_\MoveOutMethodCallInsideIfConditionRector\Fixture;
+
+use Rector\Core\Configuration\Option;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+abstract class AbstractCommand extends Command
+{
+    public function run(InputInterface $input, OutputInterface $output): int
+    {
+        return 0;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        if ($input->getOption(Option::OPTION_DEBUG)) {
+            return;
+        }
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Php\Tests\Rector\If_\MoveOutMethodCallInsideIfConditionRector\Fixture;
+
+use Rector\Core\Configuration\Option;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+abstract class AbstractCommand extends Command
+{
+    public function run(InputInterface $input, OutputInterface $output): int
+    {
+        return 0;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $inputGetOption = $input->getOption(Option::OPTION_DEBUG);
```

Comment:
This case should consider the part `::OPTION_DEBUG`, e.g.

```suggestion
        $optionDebug = $input->getOption(Option::OPTION_DEBUG);
```

Reply:
It seems not catched by `$this->expectedNameResolver->resolveForCall($methodCall)` like my comment at https://github.com/rectorphp/rector/pull/4457#issuecomment-713087908 , I wil re-check

Reply:
Yea, I think this must be added extra

Reply:
Yea, I think this must be added extra

Reply:
It seems it because `$input->getOption()` is returning `PHPStan\Type\UnionType` :

```php
^ PHPStan\Type\UnionType^ {#80356
  -types: array:4 [
    0 => PHPStan\Type\ArrayType^ {#80347
      -keyType: PHPStan\Type\MixedType^ {#80352
        -isExplicitMixed: false
        -subtractedType: null
      }
      -itemType: PHPStan\Type\StringType^ {#80348}
    }
    1 => PHPStan\Type\BooleanType^ {#80354}
    2 => PHPStan\Type\StringType^ {#80353}
    3 => PHPStan\Type\NullType^ {#37016}
  ]
}
```

So, the :

```php
$this->propertyNaming->getExpectedNameFromType($returnedType)
```

returns null on calling:

```php
$this->typeUnwrapper->unwrapNullableType($type)
```

So, it it seems need the different handling.

Reply:
implemented.